# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/crashtests/1644819.html
# Commit: 9b15baab6f77
# Full Hash: 9b15baab6f77d09eca2956276f66fee97555e4a1
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 3: Make `HTMLEditor::CollapseSelectionToEndOfLastLeafNodeOfDocument` and `HTMLEditor::InitEditorContentAndSelection` do nothing if the document is partially editable r=m_kato
#   
#   They and their callees work with the result of `GetRoot()` which is the document
#   element or the body element.  If the body is not editable, `Selection` should
#   not be updated in non-editable region nor `<br>` elements should not be
# ==============================================================================

diff -r 4d1e93c629da -r 9b15baab6f77 layout/generic/crashtests/1644819.html
--- a/layout/generic/crashtests/1644819.html	Thu Sep 22 06:24:38 2022 +0000
+++ b/layout/generic/crashtests/1644819.html	Thu Sep 22 06:27:37 2022 +0000
@@ -10,10 +10,14 @@
 </style>
 <script>
 window.onload = () => {
-  document.execCommand("selectAll", false)
-  document.execCommand("selectAll", false)
-  document.execCommand("backColor", false, "r")
-  document.execCommand("superscript", false)
+  getSelection().collapse(
+    document.querySelector("dl"),
+    document.querySelector("dl").childNodes.length
+  );
+  document.execCommand("selectAll");
+  document.execCommand("selectAll");
+  document.execCommand("backColor", false, "r");
+  document.execCommand("superscript");
 }
 </script>
 <dl style="columns:1px">